### ISSUE TITLE
Change coding standard to require all new classes to be new style

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2486]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#2486|
|Type|enhancement|
|Created|2007-02-26 02:53:10Z|
|Branch|https://github.com/twisted/twisted/tree/new-style-classes-2486|

<details><summary>Searchable metadata</summary>

```
trac-id__2486 2486
type__enhancement enhancement
reporter__itamarst itamarst
priority__high high
milestone__ 
branch__branches_new_style_classes_2486 branches/new-style-classes-2486
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1172458390000000 1172458390000000
changetime__1226844379000000 1226844379000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

New-style classes are a 2.x feature.  This isn't really related to py3k.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

(In [25328]) Branching to 'new-style-classes-2486'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Added a new section in r25331, up for review.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @thijstriemstra|
|-|-|

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [25370]) Merge new-style-classes-2486: Change coding standard to require all new classes to be new style.

Author: thijs
Reviewer: therve
Fixes: [#2486](https://github.com/twisted/twisted/issues/2486)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
